api/query.py

```diff
@@ -68,14 +69,15 @@ def post(self):
                             datapoints = []
 
                             step = 1
-                            if len_serie > 20000 and len_serie < 100000:
-                                step = 5
-                            elif len_serie >= 100000 and len_serie < 1000000:
-                                step = 100
-                            elif len_serie > 1000000:
-                                step = 500
+                            if limit_datapoints != None and int(limit_datapoint) > 0:
+                                step =  len_serie / limit_datapoints
                             else:
-                                step = 1
+                                if len_serie > 20000 and len_serie < 100000:
```

Comment:
I don't understand this code's purpose

Reply:
the limit is arbitrary, when you know that there'll be a lot of point. If you don't set this arbitrary limit, the downscaling will be done in the `else`, so grafana will be able to display faster the graph. Maybe we shouldn't do that?

Reply:
Indeed we shouldn't, if there is no limit, maybe the user wants the raw data and we shouldn't get in the way.

Reply:
ok, I removed this block